packages/suite/src/interfaces.ts

```diff
@@ -30,7 +30,7 @@ export interface TestImplementation extends Test {
   children: TestImplementation[];
 }
 
-export type Action = (context: Context) => Promise<Context | void>;
+export type Action<C extends Context, R extends Context | void> = (context: C) => Promise<R>;
```

Comment:
I'm assuming you made this change because you want to use `Action`, `Step`, etc... as the parameters for the DSL methods. It's worth thinking about if it'd actually better to add separate types for these. The type parameters added here serve a purpose when used from the test builder DSL, but they do not anywhere else, and since we do reuse these types in other places, it's worth considering if the added complexity here is worth it. I think it'd be better to just copy these types into `./dsl.ts` and leave these ones as is.

Reply:
Makes sense, I move all of the parameterized types directly into the dsl.